[MXNET-1226] add Docs update for MXNet Java (#14395)
apache/incubator-mxnet: docs/install/java_setup.md

brew update
brew tap caskroom/versions

> brew cask install java8

Reviewer: What's going on here?

Author: We don't need to install openCV for Java, not any more

brew install maven
```

**Ubuntu Steps**

Please run the following lines:

> ```bash

Reviewer: So all of the dependencies for ubuntu/scala/mxnet environment are covered with this change? I'm skeptical.

Author: We no longer require user to install that bunch... Simply Java and maven would work

<dependency>
  <groupId>org.apache.mxnet</groupId>
  <artifactId>mxnet-full_2.11-osx-x86_64-cpu</artifactId>
  <version>1.4.0</version>
</dependency>
```

> The official Java Packages is released with the release of MXNet 1.4 and now available on  [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).

Reviewer: ```suggestion
The official Java Packages are released with the release of MXNet 1.4 and are now available on the [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).
```

> The official Java Packages are released with the release of MXNet 1.4 and are now available on the [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).

Reviewer 2: ```suggestion
The official Java Packages have been released as part of MXNet 1.4 and are available on the [MXNet Maven package repository](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.mxnet%22).
```
